### DISCUSSION TITLE
Unsafe rust for optimization purposes

### DISCUSSION THREAD

**Discussion by mindoodoo** [ratatui/ratatui#66] [Discussion]
This discussion is to discuss the use of unsafe for optimization purposes.  This issue was brought up with PR #22.

For the time being, we have decided in [this](https://github.com/tui-rs-revival/tui-rs-revival/discussions/63) meeting that the policy would be to not use unsafe for optimization purposes unless it fixes **reprocible slow downs.** However, we would love to continue to discuss this issue further.

**Comment by parasyte** [comment]
This is a big old ⚠️ from me.

If you are going to accept PRs with `unsafe` code, it needs to be vetted in some way. From my observations, most projects that use `unsafe` do not have a maintainer base with the expertise required to do this vetting.

The risk/reward for performance reasons is probably not worth it unless you have a team that is very well versed in safety invariants (and breaking them with unsound code).

**Comment by GaryBeyerlein7** [comment]
So we had a discussion about the use of unsafe for optimization purposes, specifically related to PR #22. After some debate, we've decided that for now, the policy will be to avoid using unsafe for optimization unless it's necessary to fix reproducible slowdowns.

Of course, we're always open to further discussion on this topic if anyone has thoughts or concerns they'd like to share. Thanks for bringing it up!
